templates/expo-template-bare-typescript/index.js

```diff
@@ -1,7 +1,5 @@
-import { AppRegistry } from 'react-native';
+import { registerRootComponent } from 'expo';
 
 import App from './App';
-import { name as appName } from './app.json';
-import 'expo-asset';
 
-AppRegistry.registerComponent(appName, () => App);
```

Comment:
maybe it's worth adding a comment here so people have a clear understanding of what's going on and why we aren't using `AppRegistry` here

```
// registerRootComponent calls AppRegistry.registerComponent('main', () => App);
// It also handles ensuring if you load the app in Expo client or in a native build the environment is set up appropriately
```